pkg/metrics/email_domain.go

```diff
@@ -20,9 +20,16 @@ var internalDomainPattern = regexp.MustCompile(`^.*@redhat.com$|^.*@((.+)\.)?ibm
 // Note: if given email address is empty (ie, it does not exist - which should not happen),
 // then an error is logged and the returned domain is `external`
 func GetEmailDomain(obj RuntimeObject) Domain {
-	emailAddress, exists := obj.GetAnnotations()[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]
-	if !exists {
-		log.Error(nil, "no email address found in annotations", "kind", obj.GetObjectKind().GroupVersionKind().Kind, "name", obj.GetName())
+	emailAddress := ""
+
+	switch obj := obj.(type) {
+	case *toolchainv1alpha1.MasterUserRecord:
+		emailAddress = obj.Spec.PropagatedClaims.Email
+	case *toolchainv1alpha1.UserSignup:
+		emailAddress = obj.Spec.IdentityClaims.Email
+	}
```

Comment:
if you change the type of the parameter of the function to take the `PropagatedClaims`, then it would simplify the logic and it would be also clear what the function does - where the value takes from. There is no need for checking the types.
```
func GetEmailDomain(claims toolchainv1alpha1.PropagatedClaims) Domain {
	emailAddress := claims.Email
```

Reply:
This is a little simpler yes, however it would also mean that we lose details in the logs if the email address is not found.  We cannot determine the object type or the name from just the PropagatedClaims.

`log.Error(nil, "no email address found in object", "kind", obj.GetObjectKind().GroupVersionKind().Kind, "name", obj.GetName())`

Reply:
fair enough :+1: